src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -2477,7 +2435,75 @@ namespace SurfaceGeometry {
+                if (state.dataHeatBal->Zone(ZoneNum).UserEnteredFloorArea != DataGlobalConstants::AutoCalculate) {
+                    // Check entered vs calculated
+                    if (state.dataHeatBal->Zone(ZoneNum).UserEnteredFloorArea > 0.0) { // User entered zone floor area,
+                        // produce message if not near calculated
+                        if (state.dataHeatBal->Zone(ZoneNum).CalcFloorArea > 0.0) {
+                            Real64 diffp =
+                                std::abs(state.dataHeatBal->Zone(ZoneNum).CalcFloorArea - state.dataHeatBal->Zone(ZoneNum).UserEnteredFloorArea) /
+                                state.dataHeatBal->Zone(ZoneNum).UserEnteredFloorArea;
+                            if (diffp > 0.05) {
+                                ++ErrCount;
+                                if (ErrCount == 1 && !state.dataGlobal->DisplayExtraWarnings) {
+                                    ShowWarningError(state,
+                                                     std::string{RoutineName} + "Entered Zone Floor Areas differ from sum of Space Floor Area(s).");
+                                    ShowContinueError(state,
+                                                      "...use Output:Diagnostics,DisplayExtraWarnings; to show more details on individual zones.");
+                                }
+                                if (state.dataGlobal->DisplayExtraWarnings) {
+                                    // Warn user of using specified Zone Floor Area
+                                    ShowWarningError(state,
+                                                     std::string{RoutineName} + "Entered Floor Area entered for Zone=\"" +
+                                                         state.dataHeatBal->Zone(ZoneNum).Name +
+                                                         "\" significantly different from sum of Space Floor Areas");
+                                    ShowContinueError(state,
+                                                      format("Entered Zone Floor Area value={:.2R}, Sum of Space Floor Areas value={:.2R}, entered "
+                                                             "Entered Zone Floor Area will be used in calculations and Space Floor Areas will be "
+                                                             "adjusted proportionately.",
+                                                             state.dataHeatBal->Zone(ZoneNum).UserEnteredFloorArea,
+                                                             state.dataHeatBal->Zone(ZoneNum).CalcFloorArea));
+                                }
+                            }
+                        }
+                        state.dataHeatBal->Zone(ZoneNum).FloorArea = state.dataHeatBal->Zone(ZoneNum).UserEnteredFloorArea;
+                        state.dataHeatBal->Zone(ZoneNum).HasFloor = true;
+
+                        // Adjust space floor areas to match zone floor area
```

Comment:
If there is a user-input for zone floor area, always adjust the space floor areas proportionately to match. 
As I read this, maybe this should be skipped if the zone floor area value already matches the sum of the space floor areas or within some threshold?